### ISSUE TITLE
Halide build does not properly package LLVM objects on macOS

### ISSUE THREAD

**Issue by shoaibkamil** [halide/Halide#4574]
Our process for grabbing the objects from the LLVM static libraries does not correctly construct the command for some objects, resulting in not all objects being included in libHalide.a.

The below is on Xcode 11.3, macOS 10.15.2, LLVM 9.0.1 compiled from source:

```
$ g++ lesson_01*.cpp -g -I ../include -L ../lib -lHalide -o lesson_01 -std=c++11
Undefined symbols for architecture x86_64:
  "llvm::DAGTypeLegalizer::SplitRes_UNDEF(llvm::SDNode*, llvm::SDValue&, llvm::SDValue&)", referenced from:
      llvm::DAGTypeLegalizer::ExpandFloatResult(llvm::SDNode*, unsigned int) in libHalide.a(llvm_1344_LegalizeFloatTypes.cpp.o)
      llvm::DAGTypeLegalizer::SplitVectorResult(llvm::SDNode*, unsigned int) in libHalide.a(llvm_1345_LegalizeVectorTypes.cpp.o)
      llvm::DAGTypeLegalizer::ExpandIntegerResult(llvm::SDNode*, unsigned int) in libHalide.a(llvm_1346_LegalizeIntegerTypes.cpp.o)
  "llvm::DAGTypeLegalizer::ExpandRes_VAARG(llvm::SDNode*, llvm::SDValue&, llvm::SDValue&)", referenced from:
      llvm::DAGTypeLegalizer::ExpandFloatResult(llvm::SDNode*, unsigned int) in libHalide.a(llvm_1344_LegalizeFloatTypes
<many more similar errors>
```

**Comment by shoaibkamil** [comment]
I attempted the following fix:

```
diff --git a/Makefile b/Makefile
index d7c9b0f70..b17efec96 100644
--- a/Makefile
+++ b/Makefile
@@ -964,11 +964,12 @@ $(BUILD_DIR)/llvm_objects/list: $(OBJECTS) $(INITIAL_MODULES)
        # object files in which archives it uses to resolve
        # symbols. We only care about the libLLVM ones.
        @mkdir -p $(@D)
-       $(CXX) -o /dev/null -shared -Wl,-t -Wl,-t $(OBJECTS) $(INITIAL_MODULES) $(LLVM_STATIC_LIBS) $(LLVM_SYSTEM_LIBS) $(COMMON_LD_FLAGS) 2>&1 | grep "libLLVM" | grep ")" > $(BUILD_DIR)/llvm_objects/list.new
+       $(CXX) -o /dev/null -shared -Wl,-t -Wl,-t $(OBJECTS) $(INITIAL_MODULES) $(LLVM_STATIC_LIBS) $(LLVM_SYSTEM_LIBS) $(COMMON_LD_FLAGS) 2>&1 > $(BUILD_DIR)/llvm_objects/list.all
        # if the list has changed since the previous build, or there
        # is no list from a previous build, then delete any old object
        # files and re-extract the required object files
        cd $(BUILD_DIR)/llvm_objects; \
+       cat list.all | grep "libLLVM" > list.new; \
        if cmp -s list.new list; \
        then \
        echo "No changes in LLVM deps"; \
```

However, that only partially resolves the problem; some more symbols are included but not all:
```
 g++ lesson_01*.cpp -g -I ../include ../lib/libHalide.a -o lesson_01 -std=c++11
Undefined symbols for architecture x86_64:
  "llvm::DWARFDebugNames::AbbrevMapInfo::getEmptyKey()", referenced from:
      llvm::DenseMapBase<llvm::DenseMap<llvm::DWARFDebugNames::Abbrev, llvm::detail::DenseSetEmpty, llvm::DWARFDebugNames::AbbrevMapInfo, llvm::detail::DenseSetPair<llvm::DWARFDebugNames::Abbrev> >, llvm::DWARFDebugNames::Abbrev, llvm::detail::DenseSetEmpty, llvm::DWARFDebugNames::AbbrevMapInfo, llvm::detail::DenseSetPair<llvm::DWARFDebugNames::Abbrev> >::destroyAll() in libHalide.a(llvm_1380_DWARFVerifier.cpp.o)
      llvm::DenseMapIterator<llvm::DWARFDebugNames::Abbrev, llvm::detail::DenseSetEmpty, llvm::DWARFDebugNames::AbbrevMapInfo, llvm::detail::DenseSetPair<llvm::DWARFDebugNames::Abbrev>, true>::AdvancePastEmptyBuckets() in libHalide.a(llvm_1380_DWARFVerifier.cpp.o)
  "llvm::DWARFDebugNames::AbbrevMapInfo::getTombstoneKey()", referenced from:
      llvm::DenseMapBase<llvm::DenseMap<llvm::DWARFDebugNames::Abbrev, llvm::detail::DenseSetEmpty, llvm::DWARFDebugNames::AbbrevMapInfo, llvm::detail::DenseSetPair<llvm::DWARFDebugNames::Abbrev> >, llvm::DWARFDebugNames::Abbrev, llvm::detail::DenseSetEmpty, llvm::DWARFDebugNames::AbbrevMapInfo, llvm::detail::DenseSetPair<llvm::DWARFDebugNames::Abbrev> >::destroyAll() in libHalide.a(llvm_1380_DWARFVerifier.cpp.o)
      llvm::DenseMapIterator<llvm::DWARFDebugNames::Abbrev, llvm::detail::DenseSetEmpty, llvm::DWARFDebugNames::AbbrevMapInfo, llvm::detail::DenseSetPair<llvm::DWARFDebugNames::Abbrev>, true>::AdvancePastEmptyBuckets() in libHalide.a(llvm_1380_DWARFVerifier.cpp.o)
  "llvm::DWARFDebugNames::SentinelError::ID", referenced from:
      llvm::Error llvm::handleErrorImpl<llvm::DWARFVerifier::verifyNameIndexEntries(llvm::DWARFDebugNames::NameIndex const&, llvm::DWARFDebugNames::NameTableEntry const&)::$_5, llvm::DWARFVerifier::verifyNameIndexEntries(llvm::DWARFDebugNames::NameIndex const&, llvm::DWARFDebugNames::NameTableEntry const&)::$_6>(std::__1::unique_ptr<llvm::ErrorInfoBase, std::__1::default_delete<llvm::ErrorInfoBase> >, llvm::DWARFVerifier::verifyNameIndexEntries(llvm::DWARFDebugNames::NameIndex const&, llvm::DWARFDebugNames::NameTableEntry const&)::$_5&&, llvm::DWARFVerifier::verifyNameIndexEntries(llvm::DWARFDebugNames::NameIndex const&, llvm::DWARFDebugNames::NameTableEntry const&)::$_6&&) in libHalide.a(llvm_1380_DWARFVerifier.cpp.o)
<more similar errors>
```

**Comment by shoaibkamil** [comment]
Fixed by #4576.
